### PR TITLE
Validate VK_EXT_astc_decode_mode

### PR DESCRIPTION
**Pull request by sfricke-samsung**
VUID-VkImageViewASTCDecodeModeEXT-decodeMode-02230
VUID-VkImageViewASTCDecodeModeEXT-decodeMode-02231
~~VUID-VkImageViewASTCDecodeModeEXT-format-02233~~ ~~(replace with todo for future VUID)~~
VUID-VkImageViewASTCDecodeModeEXT-format-04084

### CODE REVIEW

File: layers/parameter_validation_utils.cpp [hunk 1]
```diff
@@ -900,6 +900,23 @@ bool StatelessValidation::manual_PreCallValidateCreateImageView(VkDevice device,
...
+        auto astc_decode_mode = lvl_find_in_chain<VkImageViewASTCDecodeModeEXT>(pCreateInfo->pNext);
+        if ((device_extensions.vk_ext_astc_decode_mode) && (astc_decode_mode != nullptr)) {
+            if ((astc_decode_mode->decodeMode != VK_FORMAT_R16G16B16A16_SFLOAT) &&
+                (astc_decode_mode->decodeMode != VK_FORMAT_R8G8B8A8_UNORM) &&
+                (astc_decode_mode->decodeMode != VK_FORMAT_E5B9G9R9_UFLOAT_PACK32)) {
+                skip |= LogError(device, "VUID-VkImageViewASTCDecodeModeEXT-decodeMode-02230",
+                                 "vkCreateImageView(): VkImageViewASTCDecodeModeEXT::decodeMode must be "
+                                 "VK_FORMAT_R16G16B16A16_SFLOAT, VK_FORMAT_R8G8B8A8_UNORM, or VK_FORMAT_E5B9G9R9_UFLOAT_PACK32.");
+            }
+            if (FormatIsCompressed_ASTC_LDR(pCreateInfo->format) == false) {
```
Comment (janharaldfredriksen-arm) [inline layers/parameter_validation_utils.cpp:913]: Oh, this is a spec bug...
It's supposed to be allow any ASTC format.

Comment (sfricke-samsung) [reply layers/parameter_validation_utils.cpp:913]: This is for

> format of the image view must be one of VK_FORMAT_ASTC_4x4_UNORM_BLOCK, ....

Are you saying this should be "any ASTC"  as in `SFLOAT` for HDR if `VK_EXT_texture_compression_astc_hdr` is enabled?

Comment (janharaldfredriksen-arm) [reply layers/parameter_validation_utils.cpp:913]: Yes, exactly.
I must have missed this interaction when writing the HDR extension.

Comment (sfricke-samsung) [reply layers/parameter_validation_utils.cpp:913]: ok so purpose we move ahead as follow

- Marking as WIP for now
- Will wait until spec draft is up and approved and then rewrite this PR with "UNASSIGNED_" VUID Labels and aimed to get merged in
- Update with actual VUIDs when Validation Layers updates the Vulkan Header with the upstream VUIDs generated

**Comment by sfricke-samsung** [comment]
@janharaldfredriksen-arm So I added an additional commit to add HDR format support to the format util help functions.

I would like to get your :+1: on this PR still as well

**Comment by sfricke-samsung** [comment]
Updated with `VUID-VkImageViewASTCDecodeModeEXT-format-04084` and should be ready to go now

**Comment by mark-lunarg** [comment]
Running through CI.

**Pull request merged**